illumos-utils/src/link.rs

```diff
@@ -124,7 +124,7 @@ pub struct InvalidLinkKind(String);
 /// Note that the "ownership" of the VNIC is based on convention;
 /// another process in the global zone could also modify / destroy
 /// the VNIC while this object is alive.
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```

Comment:
It's dangerous to `Clone` this without taking into account `deleted`.

With my upcoming changes in https://github.com/oxidecomputer/omicron/pull/2725 , this will result in us attempting to delete a VNIC that has already been deleted - an operation that will never succeed.

```rust
let allocator = VnicAllocator::new(...);

// Usage before this PR

// Allocates the VNIC
let mut vnic = allocator.new_control(None)?;
// De-allocates the VNIC, sets `deleted` to true.
vnic.delete();

// Usage after this PR

// Allocates the VNIC
let mut vnic = allocator.new_control(None)?;
let vnic_clone = vnic.clone();
// De-allocates the VNIC, sets `deleted` to true... but NOT for `vnic_clone`!
vnic.delete();
// This tries to destroy a VNIC that has already been deleted!
drop(vnic_clone);
```

Reply:
Good point, addressed in 8a37b046